chore(ruby): Update common options to match other platforms (#11091)
getsentry/sentry-docs: docs/platforms/ruby/common/configuration/options.mdx

---
title: Basic Options
sidebar_order: 1

> description: "Learn more about how to configure the SDK. These options are set when the SDK is first initialized, passed to the init function as an object."

Author: this copy is the same across other languages, can you fix that separately for all of them?

```ruby
config.trusted_proxies = ["2.2.2.2"]
```

</ConfigKey>

## Tracing Options

<ConfigKey name="traces-sample-rate">

A number between 0 and 1, controlling the percentage chance a given transaction will be sent to Sentry. (0 represents 0% while 1 represents 100%.) Applies equally to all transactions created in the app. Either this or <PlatformIdentifier name="traces-sampler" /> must be defined to enable tracing.

If <PlatformIdentifier name="traces-sample-rate" /> is 0, this means that no new traces will be created. However, if you have another service (for example a JS frontend) that makes requests to your service that include trace information, those traces will be continued and thus transactions will be sent to Sentry.

If you want to disable all tracing you need to set <PlatformIdentifier name="traces-sample-rate" />`=None` or set <PlatformIdentifier name="enable-tracing" />`=False`. In this case, no new traces will be started and no incoming traces will be continued.

<PlatformContent includePath="performance/traces-sample-rate" />

</ConfigKey>

<ConfigKey name="traces-sampler">

> A lambda or proc responsible for determining the percentage chance a given transaction will be sent to Sentry. It will automatically be passed information about the transaction and the context in which it's being created, and must return a number between `0` (0% chance of being sent) and `1` (100% chance of being sent). Can also be used for filtering transactions, by returning 0 for those that are unwanted. Either this or <PlatformIdentifier name="traces-sample-rate" /> must be defined to enable tracing.

Reviewer: ```suggestion
A lambda or proc that's responsible for determining the chance that a given transaction has of being sent to Sentry (from 0-100%). It will automatically be passed information about the transaction and the context in which it's being created, and must return a number between `0` (0% chance of being sent) and `1` (100% chance of being sent). 

It can also be used for filtering transactions, by returning 0 for those that are of no interest. Either this or <PlatformIdentifier name="traces-sample-rate" /> must be defined to enable tracing.
```

<PlatformContent includePath="performance/traces-sampler-as-sampler" />

</ConfigKey>

<ConfigKey name="enable-backpressure-handling">

> A boolean (default false) that controls whether a new monitor thread will be spawned to perform health checks on the SDK. If the system is unhealthy, the SDK will keep halving the `traces_sample_rate` set by you in 10 second intervals until recovery. This downsampling helps ensure that the system stays stable and reduces SDK overhead under high load.

Reviewer: ```suggestion
A boolean (the default is false) that controls whether a new monitor thread will be spawned to perform health checks on the SDK. If the system is unhealthy, the SDK will keep halving the `traces_sample_rate` set by you in 10 second intervals until recovery. This downsampling helps ensure that the system stays stable and reduces SDK overhead under high load.
```

</ConfigKey>

<ConfigKey name="trace-propagation-targets">

An optional property that controls which downstream services receive tracing data, in the form of a `sentry-trace` and a `baggage` header attached to any outgoing HTTP requests.

The option may contain an array of strings or regex against which the URLs of outgoing requests are matched.
If one of the entries in the list matches the URL of an outgoing request, trace headers will be attached to that request.

> String entries do not have to be full matches, meaning the URL of a request is matched when it _contains_ a string provided through the option.

Reviewer: ```suggestion
String entries don't have to be full matches, (meaning the URL of a request is matched when it _contains_ a string provided through the option).
```

By default, trace headers are attached to every outgoing request from the instrumented client.

</ConfigKey>

<ConfigKey name="propagate-traces">

> By default, Sentry injects `sentry-trace` and `baggage` headers to outgoing requests made with `Net::HTTP` to connect traces between services. You can disable this behavior with

Reviewer: ```suggestion
By default, Sentry injects `sentry-trace` and `baggage` headers to outgoing requests made with `Net::HTTP` to connect traces between services. You can disable this behavior with:
```

```ruby
config.propagate_traces = false
```

</ConfigKey>

## Hooks

> These options can be used to hook the SDK in various ways to customize the reporting of events.

Reviewer: ```suggestion
The below options can be used to hook the SDK in various ways and customize how events are being reported.
```

<ConfigKey name="before-send">

> Provide a lambda or proc that is called with an SDK-specific message or error event object, and can return a modified event object, or `nil` to skip reporting the event. This can be used, for instance, for manual PII stripping before sending.

Reviewer: ```suggestion
Provides a lambda or proc that's called with an SDK-specific message or error event object, and can return a modified event object, or `nil` to skip reporting the event. This can be used, for instance, for manual PII stripping before sending.
```

By the time <PlatformIdentifier name="before-send" /> is executed, all scope data has already been applied to the event. Further modification of the scope won't have any effect.

<PlatformContent includePath="configuration/before-send/" />

</ConfigKey>

<ConfigKey name="before-send-transaction">

> Provide a lambda or proc that is called with an SDK-specific transaction event object, and can return a modified transaction event object, or `nil` to skip reporting the event. One way this might be used is for manual PII stripping before sending.

Reviewer: ```suggestion
Provides a lambda or proc that's called with an SDK-specific transaction event object, and can return a modified transaction event object, or `nil` to skip reporting the event. One way this might be used is for manual PII stripping before sending.
```

Author: this isn't correct because the option doesn't do the providing, the user does

<PlatformContent includePath="configuration/before-send-transaction/" />

</ConfigKey>

<ConfigKey name="backtrace-cleanup-callback">

> If you want to clean up exceptions' backtrace before it's sent to Sentry, you can specify a callback with `backtrace_cleanup_callback` to do that. For example:

Reviewer: ```suggestion
If you want to clean up the backtrace of an exception before it's sent to Sentry, you can specify a callback with `backtrace_cleanup_callback`, for example:
```
